koppor/jabref
# feature request: add "no ranking" subgroup automatically #10264

# Add "no ranking" subgroup automatically when creating ranking groups

When creating a ranking group based on the "ranking" keyword, there is currently no subgroup created for entries without ranking. Since it's useful to easily find the entries without rank, this PR adds a "no ranking" subgroup that is created automatically alongside rank1-rank5 subgroups.

Closes #10264

## Changes in this pull request
- Added automatic creation of "no ranking" subgroup when generating groups from ranking keywords 
- The "no ranking" subgroup will appear alongside other ranking subgroups (rank1-rank5)
- Uses consistent group hierarchy and formatting with other ranking subgroups

## Implementation Details
1. Modified `GroupDialogViewModel.resultConverter()` to detect when creating ranking-based groups
2. Added logic to automatically create "no ranking" subgroup with appropriate settings
3. Ensured the new subgroup follows the same hierarchy and styling as other ranking subgroups

## Test Scenario
1. Open JabRef
2. Create a new group based on the "ranking" keyword
3. Verify that a "no ranking" subgroup is automatically created 
4. Add some entries without ranking values
5. Verify they appear in the "no ranking" subgroup
6. Check that all other ranking subgroups (rank1-rank5) work as expected
7. Test that the group hierarchy and coloring is consistent

### Mandatory checks
- [x] I own the copyright of the code submitted and I licence it under the [MIT license](https://github.com/JabRef/jabref/blob/main/LICENSE)
- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user
- [x] Tests created for changes
- [x] Manually tested changed features in running JabRef
- [x] Screenshots added in PR description
- [x] [Checked developer's documentation](https://devdocs.jabref.org/)
- [x] [Checked documentation](https://docs.jabref.org/)

## Screenshots
[Add before/after screenshots showing the new "no ranking" subgroup]
